src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/Transmission/OtlpExporterTransmissionHandler.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+using OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.ExportClient;
+using OpenTelemetry.Internal;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation.Transmission;
+
+internal class OtlpExporterTransmissionHandler<TRequest>
+{
+    public OtlpExporterTransmissionHandler(IExportClient<TRequest> exportClient)
+    {
+        Guard.ThrowIfNull(exportClient);
+
+        this.ExportClient = exportClient;
+    }
+
+    protected IExportClient<TRequest> ExportClient { get; }
+
+    /// <summary>
+    /// Sends export request to the server.
+    /// </summary>
+    /// <param name="request">The request to send to the server.</param>
+    /// <returns>True if the request is sent successfully or else false.</returns>
+    public bool TrySubmitRequest(TRequest request)
+    {
+        try
+        {
+            var response = this.ExportClient.SendExportRequest(request);
+            if (response.Success)
+            {
+                return true;
+            }
+
+            return this.OnSubmitRequestFailure(request, response);
+        }
+        catch (Exception ex)
+        {
+            OpenTelemetryProtocolExporterEventSource.Log.SubmitRequestException(ex);
+            this.OnRequestDropped(request);
+            return false;
+        }
+    }
+
+    /// <summary>
+    /// Attempts to shutdown the transmission handler, blocks the current thread
+    /// until shutdown completed or timed out.
+    /// </summary>
+    /// <param name="timeoutMilliseconds">
+    /// The number (non-negative) of milliseconds to wait, or
+    /// <c>Timeout.Infinite</c> to wait indefinitely.
+    /// </param>
+    /// <returns>
+    /// Returns <see langword="true" /> if shutdown succeeded; otherwise, <see
+    /// langword="false" />.
+    /// </returns>
+    public bool Shutdown(int timeoutMilliseconds)
+    {
+        this.OnShutdown();
```

Comment:
This seems odd. The `Shutdown` method takes in a parameter for `timeoutMilliseconds` to decide how long should it wait for before returning. Shouldn't `OnShutdown` method also take in `timeoutMilliseconds` parameter?

Reply:
Good catch. Fixed.